src/lib/support/SmartPointerUtility.h

```diff
@@ -0,0 +1,30 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <utility>
+
+/**
+ *  @brief Wrap a object which is disposable using a deleter into a unique_ptr.
+ */
+template <typename T, typename Deleter>
+auto wrap_unique(T * object, Deleter && deleter)
```

Comment:
This isn't very helpful for holding references to glib types in objects where we would want a named type for the field. I believe it would be better to actually define named wrapper types, similar to UniqueDBusConnection removed below.